File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        double successRate = this.successRate(speed);
        int carsProducedPerHour = 221;

        return speed * carsProducedPerHour * successRate;
    }

    private double successRate(int speed) {
        return switch (speed) {
            case 1, 2, 3, 4 -> 1;
            case 5, 6, 7, 8 -> 0.9;
            case 9 -> 0.8;
            case 10 -> 0.77;
            default -> 0.0;
        };
    }

    public int workingItemsPerMinute(int speed) {
        double carsProducedPerMinute = this.productionRatePerHour(speed) / 60;
        return (int) carsProducedPerMinute;
    }
 
}

File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    private const int CARS_PRODUCED_PER_HOUR = 221;

    public static double SuccessRate(int speed) =>
        speed switch
        {
            < 1 => 0,
            <= 4 => 1,
            <= 8 => 0.9,
            9 => 0.8,
            10 => 0.77,
            _ => 0
        };

    public static double ProductionRatePerHour(int speed)
    {
        return speed * CARS_PRODUCED_PER_HOUR * SuccessRate(speed);
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)(ProductionRatePerHour(speed) / 60);
    }
}
